driver.go

```diff
@@ -632,35 +618,85 @@ func memoryInBytes(strmem string) (int64, error) {
 	}
 }
 
-func parseImage(image string) (string, error) {
-	// strip http/https and docker transport prefix
-	for _, prefix := range []string{"http://", "https://", "docker://"} {
-		if strings.HasPrefix(image, prefix) {
-			image = strings.Replace(image, prefix, "", 1)
+// Creates the requested image if missing from storage
+// returns the 64-byte image ID as an unique image identifier
+func (d *Driver) createImage(image string) (string, error) {
+	var imageID string
+	imageName := image
+	// If it is a shortname, we should not have to worry
+	// Let podman deal with it according to user configuration
+	if !shortnames.IsShortName(image) {
+		imageRef, err := parseImage(image)
+		if err != nil {
+			return imageID, fmt.Errorf("invalid image reference %s: %w\n", image, err)
 		}
+		// If its non-docker transport, we cannot ask for a pull or
+		// check for existence in the API without image plumbing
+		if imageRef.Transport().Name() != "docker" {
+			if imageRef.Transport().Name() != "oci-archive" &&
+				imageRef.Transport().Name() != "docker-archive" {
+				return imageID, fmt.Errorf("unssupported transport: %s", imageRef.Transport().Name())
+			}
+			// docker-archive:path[:docker-reference]
+			// oci-archive:path:tag
+			path := strings.Split(image, ":")[1]
```

Comment:
The value for `image` is user supplied, right? If so, I think we should length check to avoid a panic

Reply:
The check is not needed because the image already went through `parseImage`. Therefore, we know for sure that in the case in which the transports are identified as `oci-archive` or `docker-archive`, the image was properly parsed by `ociArchive.ParseReference` or `dockerArchive.ParseReference`. Thus, it should be safe. 

However, I understand it is not so intuitive. The other possible solution would be:
```suggestion
			archiveData := imageRef.StringWithinTransport()
			path := strings.Split(archiveData, ":")[0]
```
Although I need to finish the archive tests first to verify this works as intended.

Reply:
Ah gotcha, missed that this morning, makes sense! I do like the suggestion you provided

Reply:
It's fixed now